packages/p2p-v2/src/hooks/useExtendedOrderDetails.ts

```diff
@@ -0,0 +1,298 @@
+import { TOrders, TServerTime } from 'types';
+import { BUY_SELL, ORDERS_STATUS } from '@/constants'; // Update your import path
+import { convertToMillis, getFormattedDateString, toMoment } from '@/utils';
+
+type TOrder = TOrders[number];
+
+type TUserDetails = TOrder['advertiser_details'] | TOrder['client_details'];
+
+type TObject = Record<string, string>;
+interface ExtendedOrderDetails extends TOrder {
+    counterpartyAdStatusString: TObject;
+    hasReviewDetails: boolean;
+    hasTimerExpired: boolean;
+    isActiveOrder: boolean;
+    isBuyOrder: boolean;
+    isBuyOrderForUser: boolean;
+    isBuyerCancelledOrder: boolean;
+    isBuyerConfirmedOrder: boolean;
+    isCompletedOrder: boolean;
+    isDisputeCompletedOrder: boolean;
+    isDisputeRefundedOrder: boolean;
+    isDisputedOrder: boolean;
+    isExpiredOrOngoingTimerExpired: boolean;
+    isExpiredOrder: boolean;
+    isFinalisedOrder: boolean;
+    isInactiveOrder: boolean;
+    isIncomingOrder: boolean;
+    isMyAd: boolean;
+    isOngoingOrder: boolean;
+    isOrderReviewable: boolean;
+    isPendingOrder: boolean;
+    isRefundedOrder: boolean;
+    isSellOrder: boolean;
+    labels: TObject;
+    myAdStatusString: TObject;
+    orderExpiryMilliseconds: number;
+    otherUserDetails: TUserDetails;
+    purchaseTime: string;
+    remainingSeconds: number;
+    shouldHighlightAlert: boolean;
+    shouldHighlightDanger: boolean;
+    shouldHighlightDisabled: boolean;
+    shouldHighlightSuccess: boolean;
+    shouldShowCancelAndPaidButton: boolean;
+    shouldShowComplainAndReceivedButton: boolean;
+    shouldShowLostFundsBanner: boolean;
+    shouldShowOnlyComplainButton: boolean;
+    shouldShowOnlyReceivedButton: boolean;
+    shouldShowOrderFooter: boolean;
+    shouldShowOrderTimer: boolean;
+    statusForBuyerConfirmedOrder: string;
+    statusForPendingOrder: string;
+    statusString: string;
+}
+
+const useExtendedOrderDetails = ({
```

Comment:
what is this hook for?

Reply:
since there are lot of function calculated results to be present in the orders page and order details page which are reusable, i thought it would be better to calculate it from the hook, and return it so that those can be directly used as values in the list and details.

Reply:
its similar to the one we have in current V1 @farrah-deriv since we manipulate the data quite a lot in order to show what we want for order-details